User service

Mefjuu94 opened the pull request:
Created Classes:
* "ValidationService" -> has two method to validate User input:
- "onlyDigits" -> return true/false 
- "onlyAlphabets" -> return true/false

* "UserService" -> has eight methods at all, and six of it is to gatherer information from user:
- "getCharacterName"
- "chooseRace"
- "chooseClass"
- "featuresOfClass"
- "chooseSpells"
- "backStory"
rest two methods are:
- "summary" - > just print summary
- "ifPharseProperly" -> checks if you can correctly parse a string of characters and convert it to digits.

*TestClasses was created to validate these classes, which contain a total of 12 tests. 6 tests per class.

- Improved mapping of Features and Spell classes, now they correctly return an array of objects instead of not one object with a string of characters.
- added "name" fields in Race class and CharacterClass

Mefjuu94 committed:
Created UserService and InputService who help with validation user input who maintain create of character:
- get character name by user input
- get character Race by user input
- get character Class by user input
- get character spells by user input if avaiable
- get character backstory by user input

Mefjuu94 committed:
-MapClass Features and MapSpells has Modified-> previously it returned a List with one item which as "name" had all attributes instead of single ones (the whole scaled string), Now it correctly creates a list of separate objects.
-Created UserServiceTests with mutiple mock answers to create posibility to UNhappy and Happy path at once to test coverage as many as possible lines of code.
- erase some empty spaces.

Mefjuu94 committed:
-InputService has been renamed to "ValidationService"
-Changed method to static in ValidationService
-ValidationService and UserService Test has been splited
-Addded ClassName in CharacterClass and RaceName in Race classes variables to get exacly name of each.
-Changed "featureList" Array in UserService class to be final.

Mefjuu94 committed:
-included adding additional fields in Race and CharacterClass in tests. Now tests pass.

Mefjuu94 committed:
-Also correct ouitput in MapRaceTests and MapCharacterTests. Now tests passes.

blazejprzyluski commented on src/main/java/org/example/Character/Mappers/MapFeatures.java:
```diff
@@ -15,8 +15,10 @@ public List<Feature> mapClassFeatures(String JSON){
             List <Feature> featureList = new ArrayList<>();
-            Feature feature = new Feature(jsonNode.get("results").findValuesAsText("name").toString());
-            featureList.add(feature);
+            Object[] feature = jsonNode.get("results").findValuesAsText("name").toArray();
```
Does this return a full list or a long string?

Mefjuu94 replied:
Return list of objects.
I can also change this:
featureList.add(new Feature(jsonNode.get("results").findValuesAsText("name").toString()));
result the same!

Mefjuu94 committed:
- changed the "Name" fields to the more precise names they refer to like "spellName", "className" ect.
- changed the record of adding objects in mapClassFeatures to be more compact.
- deleted unnecessary prints
- deleted empty lines

Mefjuu94 committed:
*Created FileSaveClass with methods:
- checkIfFileOrCharacterExist -> return true/false if Character name with same name exist.
- createCharacterDescription -> returns a String from the next method it is associated with:
- makePreetyString -> creates string without curly backets, semicolons ect.
- saveCharacter -> should create file with name of character, in specific directory.

*Created Tests for FileSaveClass.

Mefjuu94 committed:
* finished the FileSaver class:
- makePrettyString now creates clean description of Character
- changed "saveCharacter" to return boolean
* tests were written to FilSaver class.
* mapClassFeatures
- the previous version was reverted because it did not work as intended. Now correctly lists separate objects.

- With change Name of specific fields like "featureName" ect. also changed the output in tests.
